src/common/hooks/useCountdown/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect, useCallback } from "react";
+
+export const useCountdown = (): {
+  secondsLeft: number | undefined;
+  startCountdown: (seconds: number) => void;
+} => {
+  const [secondsLeft, setSecondsLeft] = useState<number | undefined>(undefined);
+
+  useEffect(() => {
+    if (
+      secondsLeft === undefined ||
+      (secondsLeft !== undefined && secondsLeft <= 0)
+    ) {
+      return;
+    }
+    const timeout = setTimeout(() => {
```

Comment:
Will it be better to not keep this running even when the number is zero? Seemed to me it will eat up resource for nothing. Thinking we can clear the timeout when it reaches zero and when you callback with a time it sets it again.